Examples/Algorithms/TrackFinding/src/AmbiguityResolutionAlgorithm.cpp

```diff
@@ -28,25 +26,23 @@ ActsExamples::AmbiguityResolutionAlgorithm::AmbiguityResolutionAlgorithm(
     Acts::Logging::Level lvl)
     : ActsExamples::IAlgorithm("AmbiguityResolutionAlgorithm", lvl),
       m_cfg(std::move(cfg)) {
-  if (m_cfg.inputTrajectories.empty()) {
+  if (m_cfg.inputTracks.empty()) {
     throw std::invalid_argument("Missing trajectories input collection");
   }
-  if (m_cfg.outputTrajectories.empty()) {
+  if (m_cfg.outputTracks.empty()) {
     throw std::invalid_argument("Missing trajectories output collection");
   }
-
-  m_inputTrajectories.initialize(m_cfg.inputTrajectories);
-  m_outputTrajectories.initialize(m_cfg.outputTrajectories);
+  m_inputTracks.initialize(m_cfg.inputTracks);
+  m_outputTracks.initialize(m_cfg.outputTracks);
 }
 
 namespace {
 
 struct State {
   std::size_t numberOfTracks{};
 
-  std::vector<std::pair<std::size_t, std::size_t>> trackTips;
+  std::vector<int> trackTips;
```

Comment:
This is only internal to the algorithm, right?

Reply:
Yes, I think so. @andiwand can probably comment more on this

Reply:
yes this is algorithm internal. I use this state to operate on with free functions

as far as I can tell the `IndexType` will end up as `std::uint32_t`. so `int` should be fine